google-cloud-bigquery/lib/google/cloud/bigquery/table.rb

```diff
@@ -1387,13 +1397,32 @@ def extract_job extract_url, format: nil, compression: nil,
         #
         #   table.extract "gs://my-bucket/file-name.json", format: "json"
         #
+        # @example Extract to a CSV file, attaching labels to the job:
+        #   require "google/cloud/bigquery"
+        #
+        #   bigquery = Google::Cloud::Bigquery.new
+        #   dataset = bigquery.dataset "my_dataset"
+        #   table = dataset.table "my_table"
+        #
+        #   table.extract "gs://my-bucket/file-name.csv" do |extract|
+        #     extract.labels = { "custom-label" => "custom-value" }
+        #   end
+        #
         # @!group Data
         #
         def extract extract_url, format: nil, compression: nil, delimiter: nil,
                     header: nil
-          job = extract_job extract_url, format: format,
```

Comment:
Instead of removing the call to `extract_job`, how about passing the `extract` block to it instead?

```ruby
def extract extract_url, format: nil, compression: nil, delimiter: nil,
                      header: nil, &block
  job = extract_job extract_url, format: format,
                                 compression: compression,
                                 delimiter: delimiter, header: header,
                                 &block

Reply:
Done in https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/2014/commits/45a556d39cbd6dae1e7c50fad9c9b004e443032a